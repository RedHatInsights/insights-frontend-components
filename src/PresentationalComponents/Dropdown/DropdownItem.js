import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DropdownItem as PfDropdownItem, DropdownSeparator } from '@patternfly/react-core';

class DropdownItem extends Component {
    render() {
        const {
            className,
            children,
            isDisabled,
            isSeparator,
            ...props
        } = this.props;

        const DropCmp = isSeparator ? DropdownSeparator : PfDropdownItem;

        console.warn('DropdownItem from FE component shouldn\'t be used anymore. \
Instead use http://patternfly-react.surge.sh/patternfly-4/components/dropdown#DropdownItem from PF repository.');
        return (
            <DropCmp
                { ...props }
                isDisabled={ isDisabled }
                className={
                    isSeparator ?
                        className.split(' ').filter(oneClass => oneClass !== 'pf-c-dropdown__menu-item').join(' ') :
                        className
                } >
                { !isSeparator && children }
            </DropCmp>
        );
    }
}

DropdownItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    component: PropTypes.node,
    isDisabled: PropTypes.bool
};

DropdownItem.defaultProps = {
    children: null,
    className: '',
    isDisabled: false
};

export default DropdownItem;
