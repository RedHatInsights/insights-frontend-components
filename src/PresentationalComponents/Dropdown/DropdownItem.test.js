import React from 'react';
import DropdownItem from './DropdownItem';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('DropdownItem component', () => {
    console.warn = jest.fn();
    describe('should render correctly', () => {
        it('no props', () => {
            const wrapper = shallow(<DropdownItem />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('separator', () => {
            const wrapper = shallow(<DropdownItem isSeparator className="pf-c-dropdown__menu-item fsf">test</DropdownItem>);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('disabled', () => {
            const wrapper = shallow(<DropdownItem isDisabled />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});
