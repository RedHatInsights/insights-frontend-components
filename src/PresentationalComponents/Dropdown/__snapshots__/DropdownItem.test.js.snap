// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem component should render correctly disabled 1`] = `
<Item
  className=""
  component="a"
  href="#"
  isDisabled={true}
  isHovered={false}
/>
`;

exports[`DropdownItem component should render correctly no props 1`] = `
<Item
  className=""
  component="a"
  href="#"
  isDisabled={false}
  isHovered={false}
/>
`;

exports[`DropdownItem component should render correctly separator 1`] = `
<Separator
  className="fsf"
  component="a"
  href="#"
  isDisabled={false}
  isHovered={false}
/>
`;
